deps/lua/src/ldo.c

```diff
@@ -245,8 +246,10 @@ static StkId tryfuncTM (lua_State *L, StkId func) {
   const TValue *tm = luaT_gettmbyobj(L, func, TM_CALL);
   StkId p;
   ptrdiff_t funcr = savestack(L, func);
-  if (!ttisfunction(tm))
+  if (!ttisfunction(tm)) {
+    luaD_checkstack(L, 1);
     luaG_typeerror(L, func, "call");
+  }
```

Comment:
Here I am also think that `luaD_checkstack` is not required, I could not find any flow that starts from `luaG_typeerror` and requires this extra stack element. All the code flows that push data to the stack calls `incr_top` which will call `luaD_checkstack` (notice that this is a common pattern on the Lua code, first push the element and only after check the stack and increase if needed. This pattern is possible because of the fact that Lua always keeps 5 extra elements on top of the Lua stack top).